### PR TITLE
feat: add option to enrich an entity 

### PR DESCRIPTION
**Pull request by juliannemarik**
[7621](https://devtopia.esri.com/dc/hub/issues/7621)

### Description
- make `toEditor` in `IWithEditorBehavior` and all associated class methods async
- add new arg, `include`, to the `toEditor` class method
- add `enrichEntity` util that delegates to helper functions to add async enrichments to the entity. For now, only a followers group enrichment is supported. `enrichEntity` gets called from all `toEditor` class methods with the array of `include` strings provided

1. [x] Updated meaningful TSDoc to methods including Parameters and Returns, see [Documentation Guide](https://esri.github.io/hub-components/storybook/?path=/story/guides-documentation--page)
1. [x] used semantic commit messages
1. [x] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)
1. [x] updated `peerDependencies` as needed. **CRITICAL** our automated release system can **not** be counted on to update `peerDependencies` so we _must_ do it manually in our PRs when needed. See the [updating peerDependencies](/RELEASE.md#Updating-peerDependencies) section of the release instructions for more details.

### CODE REVIEW

File: packages/common/src/core/schemas/internal/enrichEntity.ts
Comment (juliannemarik) [inline packages/common/src/core/schemas/internal/enrichEntity.ts:1]: wondering where this should live - should it live here within the `schemas` directory and only be leveraged in the context of `IWithEditorBehavior`, or should it live up a level and be a util that can be called from anywhere to enrich an entity with various information?

File: packages/common/src/core/schemas/internal/enrichEntity.ts
Comment (dbouwman) [inline packages/common/src/core/schemas/internal/enrichEntity.ts:59]: Maybe add a comment in here re: why we chose not to do the whole OperationPipeline thing and/or note that it as intentional

File: packages/common/src/core/schemas/internal/enrichEntity.ts
Comment (dbouwman) [inline packages/common/src/core/schemas/internal/enrichEntity.ts:85]: what should this do it `followersGroupId` is null? letting getGroup throw in the middle of this will be messy

**Review by dbouwman — APPROVED** [review]
This all looks good to me - added a few comments: re adding more comments in the code, but otherwise 👍🏻. I think Tom should weigh in re: alignment / connection to the content enrichments - we want this to be conceptually the same, and even better if it could interoperate.

File: packages/common/src/content/HubContent.ts
Comment (tomwayson) [inline packages/common/src/content/HubContent.ts:118]: I'd like for @sonofflynn89 to take a look at this (after you merge) when he gets back and consider whether or not it would make sense to use this as the point to fetch the server enrichment for feature service content.

File: packages/common/src/initiatives/HubInitiative.ts
Comment (tomwayson) [inline packages/common/src/initiatives/HubInitiative.ts:262]: I was sort of expecting to see each entity type provide it's own list of default `include`s. So the default for initiatives (or sites) would be `['followersGroup']`. That way the default behavior of `initiativeEntity.toEditor()` would be to include the followers group information. You could override that default behavior by passing `[]` as the `include` argument.

**Review by tomwayson — APPROVED** [review]
This looks good.

It will be good to have this second point in the workflow to fetch enrichments. One thing to watch out for is if devs are confused as to whether it should be done in the initial fetch, or here.

File: packages/common/src/initiatives/HubInitiative.ts
Comment (juliannemarik) [reply packages/common/src/initiatives/HubInitiative.ts:262]: Leaving this here for posterity even though we discussed in component standup: rather than having the entities themselves define defaults, defaults are being defined/passed in at the ui level (e.g. from the `arcgis-hub-entity-editor`). This is because the defaults differ depending on the ui-context we're in (not necessarily the entity) - on the followers pane (`editorType = "hub:<entity>:followers"`), we'll always want the `followersGroup` enrichment, but we don't need this on the details pane (`editorType = "hub:<entity>:edit"`)

**Pull request merged**
